packages/engine/lib/stateful/src/field/mod.rs

```diff
@@ -1,13 +1,23 @@
-//! Provides an hEngine-specific representation of Arrows `Field` and `DataType`.
+//! Provides an hEngine-specific representation of Arrows `Field`, `DataType`, and `Schema`.
```

Comment:
I'm not sure I quite agree with this, otherwise we'd put this module under `arrow`. We're not representing Arrow Fields so much as defining our own Fields which **use** `arrow` fields in their implementation.

This module provides the ways through which we let users define `Field`s on their agents, and the `DataType`s we expose.

Reply:
I changed the description to "Provides an abstraction over Arrows `Field`, `DataType`, and `Schema`." in b1b7b08f4269775b94e6a768bb1db2015ff6030e. Also, I mentioned, that we require the custom types because we have additional constraints on our types.